### ISSUE TITLE
Sphinx search results show "404: Page not found" in preview snippets

### ISSUE THREAD

**Issue by speth** [Cantera/cantera#1692]
**Problem description**

In the Sphinx documentation for the development version, the content preview snippet for each result page is showing text from the 404 page rather than the from the page that contains the search string. The page title is correct, and this links to the correct page. This affects all search results except those to specific API documented objects, where no preview is typically given. 

**Steps to reproduce**

1. Follow this link for a search results page: https://cantera.org/dev/search.html?q=equilibrium#

**Behavior**

2. See search results:

---
![image](https://github.com/Cantera/cantera/assets/644977/bc667b23-d411-4d30-8e0d-0776bb24f067)

---

3. Note the URLs that are being accessed in the server logs:
```
- - [24/Apr/2024:16:51:14 +0000] "GET /dev/undefinedexamples/matlab_experimental/flame.html HTTP/1.1" 404 6131 "https://cantera.org/dev/search.html?q=equilibrium" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:124.0) Gecko/20100101 Firefox/124.0"
```

Note the substring `undefined` which is incorrectly being injected into the URL.

This does not occur with the current "stable" / 3.0 documentation. For example, see https://cantera.org/documentation/docs-3.0/sphinx/html/search.html?q=equilibrium#.

**System information**

- Cantera version: `main` branch at d37a76bf6

**Comment by speth** [comment]
Using the copy of the docs that hosted at `testing.cantera.org`, which I think will be representative of how things are set up after the transition, the behavior is slightly different. There are no longer any 404 errors, and Sphinx reads all the relevant pages, but then doesn't show any preview snippets. See https://testing.cantera.org/stable/search.html?q=equilibrium#, for example.

**Comment by speth** [comment]
Aha! the version of this affecting `testing.cantera.org` is this due to https://github.com/pydata/pydata-sphinx-theme/issues/1676, which should be fixed if we update to `pydata-sphinx-theme` 0.15.2 or newer.
